src/pandapipes/component_models/heat_consumer_component.py

```diff
@@ -0,0 +1,231 @@
+# Copyright (c) 2020-2023 by Fraunhofer Institute for Energy Economics
+# and Energy System Technology (IEE), Kassel. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+import numpy as np
+from numpy import dtype
+
+from pandapipes.component_models import get_fluid, \
+    BranchWZeroLengthComponent, get_component_array, standard_branch_wo_internals_result_lookup
+from pandapipes.component_models.junction_component import Junction
+from pandapipes.idx_branch import (D, AREA, VINIT, ALPHA, QEXT, RHO, TEXT, JAC_DERIV_DP1,
+                                   JAC_DERIV_DV, JAC_DERIV_DP, LOAD_VEC_BRANCHES)
+from pandapipes.pf.result_extraction import extract_branch_results_without_internals
+
+
+class HeatConsumer(BranchWZeroLengthComponent):
+    """
+
+    """
+    # columns for internal array
+    MASS = 0
+    QEXT = 1
+    DELTAT = 2
+    TRETURN = 3
+    MODE = 4
+
+    internal_cols = 5
+
+    # numbering of given parameters (for mdot, qext, deltat, treturn)
+    MF = 0
+    QE = 1
+    DT = 2
+    TR = 3
+
+    # heat consumer modes (sum of combinations of given parameters)
+    MF_QE = 1
+    MF_DT = 2
+    MF_TR = 4
+    QE_DT = 3
+    QE_TR = 5
+
+    @classmethod
+    def table_name(cls):
+        return "heat_consumer"
+
+    @classmethod
+    def get_connected_node_type(cls):
+        return Junction
+
+    @classmethod
+    def from_to_node_cols(cls):
+        return "from_junction", "to_junction"
+
+    @classmethod
+    def active_identifier(cls):
+        return "in_service"
+
+    @classmethod
+    def create_pit_branch_entries(cls, net, branch_pit):
+        """
+        Function which creates pit branch entries with a specific table.
+        :param net: The pandapipes network
+        :type net: pandapipesNet
+        :param branch_pit:
+        :type branch_pit:
+        :return: No Output.
+        """
+        hs_pit = super().create_pit_branch_entries(net, branch_pit)
+        hs_pit[:, D] = net[cls.table_name()].diameter_m.values
+        hs_pit[:, AREA] = hs_pit[:, D] ** 2 * np.pi / 4
+        hs_pit[:, VINIT] = (net[cls.table_name()].controlled_mdot_kg_per_s.values /
+                            (hs_pit[:, AREA] * hs_pit[:, RHO]))
+        hs_pit[:, ALPHA] = 0
+        hs_pit[:, QEXT] = net[cls.table_name()].qext_w.values
+        hs_pit[:, TEXT] = 293.15
```

Comment:
I think we should move this up to the BranchWZeroLengthComponent

Reply:
Agreed, I moved text and alpha both for heat consumer and heat exchanger.